draft-ietf-quic-transport.md

```diff
@@ -1461,127 +1462,215 @@ connection if the integrity check fails with a PROTOCOL_VIOLATION error code.
+the causative PATH_CHALLENGE.
+
+When the client receives a PATH_RESPONSE from the server and verifies that the
+included data matches that sent by the client in a previously sent
+PATH_CHALLENGE, the server is considered reachable from the new address and the
+PMTU of the new network path is considered verified.
+
+A client SHOULD abandon the new address if it does not receive a PATH_RESPONSE
+after sending some number of PATH_CHALLENGE frames and/or after some time has
+passed. Note that the client may be receiving data from the server during this
+period, but not receiving a PATH_RESPONSE may be indicative of a PMTU detection
+failure.
+
+
+#### Responding to Server's Validation
+
+A client may receive a PATH_CHALLENGE from a server that is validating the
+client's ownership of its address, as described in {{migration-server}}. The
+client MUST respond to this frame by echoing the data from the server's
+PATH_CHALLENGE frame in a PATH_RESPONSE frame.
+
+A client's PATH_RESPONSE MAY be bundled with other frames and does not require
+additional padding.
+
+A client MAY send more than one PATH_RESPONSE in response to a PATH_CHALLENGE,
+as allowed by the congestion controller, to ensure that packet loss does not
+unduly delay or cause failure of the server's validation process.
+
+A client may trigger the server's validation unwittingly when its public address
+changes, such as when a NAT rebinds the client's private address to a new public
+address.  Though the client may be oblivious of such an address change, it MUST
+respond to the consequent validation.  Receiving a PATH_CHALLENGE from the
+server when the client is not actively changing addresses may be indicative of
+other network activity causing its public address to change, or it may be
+indicative of a server choosing to validate a client's address after a period of
+quiescence.
+
+
+### Server Behavior {{migration-server}}
+
+The server responds to any probes sent by the client from a new address,
+participates in the client's PMTU verification, and MUST validate that the
+client owns its new address.  The server MAY validate any client's address at
+any time, but it MUST limit the rate at which data is sent to an unvalidated
+address, and it MUST limit the amount of time for which it sends data to an
+unvalidated address.
+
+
+#### Responding to Client's Probe
+
+When a server receives a PATH_CHALLENGE frame from a client, it MUST generate a
+PATH_RESPONSE frame that contains the same data as the challenge, and send it in
+a packet padded to the size of the client's packet that included the
+PATH_CHALLENGE. This PATH_RESPONSE MUST be sent to the address from which the
+PATH_CHALLENGE was received.
+
+The server need not retransmit its PATH_RESPONSE, since it is the client's
+responsibility to send more PATH_CHALLENGE frames when a response is not
+received for a prior one.
+
+
+#### Validating an Unvalidated Client Address
```

Comment:
Let's call this `{#migration-validate}` since we refer to that elsewhere in the document?

Reply:
done